inc/class-tachyon.php

```diff
@@ -788,4 +824,74 @@ protected static function image_sizes() {
 
 		return is_array( self::$image_sizes ) ? self::$image_sizes : array();
 	}
+
+
+	/**
+	 * Determine if image_downsize should utilize Tachyon via REST API.
+	 *
+	 * The WordPress Block Editor (Gutenberg) and other REST API consumers using the wp/v2/media endpoint, especially in the "edit"
+	 * context is more akin to the is_admin usage of Tachyon (see filter_image_downsize). Since consumers are trying to edit content in posts,
+	 * Tachyon should not fire as it will fire later on display. By aborting an attempt to change an image here, we
+	 * prevents issues like https://github.com/Automattic/jetpack/issues/10580
+	 *
+	 * To determine if we're using the wp/v2/media endpoint, we hook onto the `rest_request_before_callbacks` filter and
+	 * if determined we are using it in the edit context, we'll false out the `tachyon__override_image_downsize` filter.
+	 *
+	 * @author JetPack Photo / Automattic
+	 * @param null|WP_Error   $response
```

Comment:
Missing parameter descriptions here
```suggestion
	 * @param null|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
```